pep517/compat.py

```diff
@@ -36,16 +36,21 @@ def read_json(path):
 
 
 if sys.version_info < (3, 6):
-    from toml import load as _toml_load  # noqa: F401
+    from toml import load as _toml_load_str  # noqa: F401
 
-    def toml_load(f):
+    def _toml_load_bytes(f):
         w = io.TextIOWrapper(f, encoding="utf8", newline="")
         try:
-            return _toml_load(w)
+            return _toml_load_str(w)
         finally:
             w.detach()
 
     from toml import TomlDecodeError as TOMLDecodeError  # noqa: F401
 else:
-    from tomli import load as toml_load  # noqa: F401
+    from tomli import load as _toml_load_bytes  # noqa: F401
     from tomli import TOMLDecodeError  # noqa: F401
+
+
+def read_toml(path):
```

Comment:
My thinking when I suggested this was that we would define this function twice *instead of* fiddling with TextIOWrapper. I was thinking of it as a simplification (admittedly a very minor one), rather than an extra layer of abstraction.